src/redux/cities/cities.js

```diff
@@ -0,0 +1,112 @@
+  },
+  {
+    cityName: 'Paris',
+    cityImage: 'https://media.cntraveler.com/photos/5cf96a9dd9fb41f17ed08435/master/pass/Eiffel%20Tower_GettyImages-1005348968.jpg',
+  },
+  {
+    cityName: 'London',
+    cityImage: 'https://a.cdn-hotels.com/gdcs/production27/d274/43014cca-c88c-4061-ace8-58edc24531ee.jpg?impolicy=fcrop&w=800&h=533&q=medium',
+  },
+  {
+    cityName: 'Madrid',
+    cityImage: 'https://media.istockphoto.com/photos/madrid-spain-on-gran-via-picture-id1297090032?k=20&m=1297090032&s=170667a&w=0&h=Qa_DyLiwIcrWwiV_jQp9pLM2yCDuymqwveCbLOsdWjY=',
+  },
+  {
+    cityName: 'Barcelona',
+    cityImage: 'https://images.unsplash.com/photo-1583422409516-2895a77efded?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8YmFyY2Vsb25hJTIwY2l0eXxlbnwwfHwwfHw%3D&w=1000&q=80',
+  },
+  {
+    cityName: 'Berlin',
+    cityImage: 'https://media.istockphoto.com/photos/berlin-skyline-with-spree-river-at-sunset-germany-picture-id486585530?k=20&m=486585530&s=612x612&w=0&h=FnNkM2BgTNv5-WQUpT2iB5R_0YSeW32RnLP7FYzOVVY=',
+  },
+  {
+    cityName: 'Milan',
+    cityImage: 'https://media.timeout.com/images/105186767/750/422/image.jpg',
+  },
+  {
+    cityName: 'El cairo',
+    cityImage: 'https://media.informabtl.com/wp-content/uploads/2019/05/a8cac080-el-cairo.jpg',
+  },
+  {
+    cityName: 'Beijing',
+    cityImage: 'https://www.tripsavvy.com/thmb/LkZ9tuEaT2CgpFUJRZ0R73yJbh0=/2119x1192/smart/filters:no_upscale()/beijing-central-business-district-buildings-skyline--china-cityscape-477749918-ddee5166b7474fd881b53f87244cc6d2.jpg',
+  },
+  {
+    cityName: 'Tokio',
+    cityImage: 'https://ep01.epimg.net/elviajero/imagenes/2020/04/30/actualidad/1588237365_988791_1588237524_noticia_normal.jpg',
+  },
+  {
+    cityName: 'Seoul',
+    cityImage: 'https://www.webuildvalue.com/wp-content/uploads/seoul-megacity-crescita-sostenibile.jpg',
+  }];
+  const cities = await Promise.all(listOfCities.map(async (city) => {
+    const searchName = city.cityName.replace(/ /g, '+');
+    const rawCityData = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${searchName}&appid=5cbd72262755db9398286c3cd58a219a&units=imperial`);
+    const cityData = await rawCityData.json();
+    // new city Props
+    const id = cityData.id;
+    const icon = `http://openweathermap.org/img/wn/${cityData.weather[0].icon}@2x.png`;
+    const mainDesc = cityData.weather[0].main;
+    const description = cityData.weather[0].description;
+    const temp = cityData.main.temp;
+    const tempFeelsLike = cityData.main.feels_like;
+    const pressure = cityData.main.pressure;
+    const humidity = cityData.main.humidity;
+    const visibility = cityData.visibility;
+    const cityCountry = cityData.sys.country;
+    const newData = {
+      id,
+      icon,
+      mainDesc,
+      description,
+      temp,
+      tempFeelsLike,
+      pressure,
+      humidity,
+      visibility,
+      cityCountry,
+    };
+    return { ...city, ...newData };
+  }));
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/163305833-8b90354e-7b2b-41ec-bf4d-3db3cf827fc9.png)

I can see you are filtering the data from the API on the redux file, not on a `Filter` stateless component.

Therefore:

- [ ] In order to meet this requirement, please, kindly ensure that you filter the data from the API on a new component called "Filter", and make sure this component is stateless.

That way, we will meet this particular requirement.

Reply:
Hey @Meltrust what exactly do you mean by a "Filter stateless" component? I don't understand this requirement. Can I see a reference or a working example relevant to mine? I don't think I'd describe what that piece of code does as "filtering".

Reply:
Hey @franklinben23, I'm looking at your inquiry.

Reply:
Hey @franklinben23, here is your answer:

This is your data from the API (in this case, New York):

![image](https://user-images.githubusercontent.com/57421823/163311829-5aa19cc1-76ac-4195-a52f-bd0ae0dfc7e3.png)

You are filtering this data because you are not using all of it.  You are extracting just the data that is needed for your project.

According to the requirement, you need to create a new component called "Filter" and do all the filtering there.  In other words, the data that you retrieve from the API should be sent there, and the filtering of the required information should happen also there.

Lastly, make sure this new "Filter" component is stateless.  As the name implies, it means that the component does not use state.

We cannot make this requirement optional, because it is listed on the project requirements, as shown in the review.

I hope this clears your concerns and makes sense to you.

If you have any more inquiries, please don't hesitate to ask.